Implement legacy mode for external laser file reading (#3923)
ECP-WarpX/WarpX: Docs/source/usage/parameters.rst

      none of the parameters below are used when ``<laser_name>.parse_field_function=1``. Even
      though ``<laser_name>.wavelength`` and ``<laser_name>.e_max`` should be included in the laser
      function, they still have to be specified as they are used for numerical purposes.

>     - ``"from_file"``: the electric field of the laser is read from an external file. Currently both Laser Symple Manipulator (lasy) and binary formats are supported
>       (see the lasy docs: https://lasydoc.readthedocs.io/en/latest/ ). It requires to provide the name of the file to load
>       setting the additional parameter ``<laser_name>.binary_file_name`` or ``<laser_name>.lasy_file_name`` (`string`). It accepts an
>       optional parameter ``<laser_name>.time_chunk_size`` (`int`) , only supported for a lasy file; this allows to read only

Reviewer: ```suggestion
    - ``"from_file"``: the electric field of the laser is read from an external file. Currently both the `lasy <https://lasydoc.readthedocs.io/en/latest/>`__ format as well as a custom binary format are supported. It requires to provide the name of the file to load
      setting the additional parameter ``<laser_name>.binary_file_name`` or ``<laser_name>.lasy_file_name`` (`string`). It accepts an
      optional parameter ``<laser_name>.time_chunk_size`` (`int`) , only supported for a lasy file; this allows to read only
```
